Test/gsapTests/ModelBasedPrognoserTests.cpp

```diff
@@ -0,0 +1,84 @@
+//
+//  ModelBasedPrognoserTests.cpp
+//  testLib
+//
+//  Created by Teubert, Christopher (ARC-TI) on 12/13/18.
+//
+
+#include <thread>
+
+#include "MockClasses.h"
+#include "ModelBasedPrognoser.h"
+#include "ModelBasedPrognoserTests.h"
+#include "Models/PrognosticsModelFactory.h"
+#include "Observers/ObserverFactory.h"
+#include "Predictors/PredictorFactory.h"
+#include "Test.h"
+
+using namespace PCOE;
+using namespace PCOE::Test;
+
+void testWithMockModel() {
+	PrognosticsModelFactory::instance().Register<TestPrognosticsModel>("Mock");
+	ObserverFactory::instance().Register<TestObserver>("Mock");
+	PredictorFactory::instance().Register<TestPredictor>("Mock");
+	ConfigMap config;
+	config.set("model", "Mock");
+	config.set("observer", "Mock");
+	config.set("predictor", "Mock");
+	config.set("LoadEstimator.Loading", {"1", "2"});
+	ModelBasedPrognoser mbp(config);
+	
+	// Initialize
+	std::map<MessageId, Datum<double> > data;
+	data[MessageId::TestInput0] = Datum<double>(1);
+	data[MessageId::TestInput1] = Datum<double>(2);
+	data[MessageId::TestOutput0] = Datum<double>(3);
+	mbp.step(data);
+	
+	std::this_thread::sleep_for(std::chrono::milliseconds(1));
```

Comment:
Why do we need to sleep? Sleeps slow down the test suite, and are usually a sign it's possible to improve some aspect of the code to make the sleep unnecessary.